pkg/workflows/management/install_new_components.go

```diff
@@ -25,6 +25,12 @@ func runInstallNewComponents(ctx context.Context, commandContext *task.CommandCo
 		commandContext.SetError(err)
 		return err
 	}
+
+	commandContext.ClusterSpec.Cluster.SetManagmentComponentsVersion(commandContext.ClusterSpec.EKSARelease.Spec.Version)
+	if err := commandContext.Client.Apply(ctx, commandContext.ManagementCluster.KubeconfigFile, commandContext.ClusterSpec.Cluster); err != nil {
```

Comment:
Why wouldn't we do this in the "upgrade management components" task?
It seems that would be a more obvious place. Since we made this change only after updating those components.

Reply:
If I understand this correctly, I think you're referring to the `upgradeCoreComponents` task. I've considered this a bit more and I agree with this point, so I've moved it there.